gtest/src/log.rs

```diff
@@ -18,8 +21,40 @@ impl CoreLog {
             dest: other.dest,
             payload: other.payload.into_raw(),
             exit_code: other.reply.map(|(_, code)| Some(code)).unwrap_or_default(),
+            message_id: other.id
         }
     }
+
+    pub(crate) fn generate_reply(&self, payload: Payload, message_id: MessageId, value: u128) -> Message{
+        Message{
+            source: self.source,
+            dest: self.dest,
+            payload,
+            gas_limit: None,
+            value,
+            id: message_id,
+            reply: match self.exit_code {
+                None => {None}
+                Some(exit_code) => {Some((self.message_id, exit_code))}
+            }
```

Comment:
Use `Option::map()`:

```suggestion
            reply: self.exit_code.map(|exit_code| (self.message_id, exit_code))
```